src/main/Security/AccountSecurityController.java

```diff
@@ -25,7 +28,7 @@ public AccountSecurityController(MongoDatabase db) {
         JSONObject req = new JSONObject(ctx.body());
         String username = req.getString("username");
         ForgotPasswordService forgotPasswordService =
-            new ForgotPasswordService(db, logger, username);
+            new ForgotPasswordService(db, getUserService, logger, username);
```

Comment:
hmm instead of passing in a getUserService, why don't we just pass in the UserDao itself? it removes a layer of abstraction, and we can remove the db input, similar to the training modules

Reply:
Agreed for the context I'm using it here but I think having some kind of layer between these services and the dao to do null checks and security checks, there's a lot of duplicate code. But I agree, the getUserService literally does nothing here